### PR TITLE
lint, improve docs contrib

### PR DESCRIPTION
**Pull request by amberisvibin**
- lint `docs/Benchmarks.mdr`, `docs/BuildingTheDocs.mdr`, `docs/Contributing/ContributingCode.mdr`, `docs/Contributing/ContributingDocumentation.mdr`
- improve `docs/Contributing/ContributingDocumentation.mdr`
- move `404.mdr`
- add links to dependencies in `docs/Contributing/ContributingDocumentation.mdr`

### CODE REVIEW

**Comment by fennecdjay** [comment]
That's pretty good :smile:
Could you provide links for the dependencies of contribocs?

File: docs/Contributing/ContributingDocumentation.mdr [hunk 1]
```diff
@@ -5,19 +5,26 @@
...
 ## Dependencies
 
-  * Gwion
-  * mdr
-  * MdBook
+Technically none of these are *needed*, but they are quite useful:
+
+* Gwion
+* mdr
+* MdBook
```
Comment (fennecdjay) [inline docs/Contributing/ContributingDocumentation.mdr]: I think this misses links.
 * gwion:  https://github.com/fennecdjay/gwion
 * mdr:    https://github.com/fennecdjay/mdr
 * mdbook: https://github.com/rust-lang/mdBook

Comment (amberisvibin) [reply docs/Contributing/ContributingDocumentation.mdr]: Okay, Ill get that in.

Comment (amberisvibin) [reply docs/Contributing/ContributingDocumentation.mdr]: Ok, done.

File: docs/Contributing/ContributingCode.mdr [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+
+``` sh
+cd gwion
+make watch
+```
+
+### Add your changes
+
+``` sh
+git add docs/myfile.mdr
```
Comment (fennecdjay) [inline docs/Contributing/ContributingCode.mdr]: Shouldn't this be more like
``` sh
git add src/myfile.c
```
?

**Comment by fennecdjay** [comment]
The naming in git add is the only thing slightly suspicious.
The rest looks good to me, great!

File: docs/Contributing/ContributingCode.mdr (hunk 2 above)
Comment (amberisvibin) [reply docs/Contributing/ContributingCode.mdr]: Also correct. I'll fix that real quick.

Comment (amberisvibin) [reply docs/Contributing/ContributingCode.mdr]: Done.

Comment (fennecdjay) [reply docs/Contributing/ContributingCode.mdr]: Nice.
Is it OK to merge for you, or is there anything left?

Comment (amberisvibin) [reply docs/Contributing/ContributingCode.mdr]: No, I think that's it. Thank you!

Comment (fennecdjay) [reply docs/Contributing/ContributingCode.mdr]: Merging now.
Nice Work :smiley:

**Pull request merged**